tests/api.rs

```diff
@@ -1951,6 +1951,6 @@ fn assert_lt(left: usize, right: usize) {
 #[test]
 fn session_types_are_not_huge() {
     // Arbitrary sizes
-    assert_lt(mem::size_of::<ServerSession>(), 1536);
-    assert_lt(mem::size_of::<ClientSession>(), 1536);
+    assert_lt(mem::size_of::<ServerSession>(), 2300);
+    assert_lt(mem::size_of::<ClientSession>(), 2300);
```

Comment:
AFAICT, this is due to storing the secrets inline as `Prk` instead of on the heap inside `Vec<u8>`s. I think this is a good change.

As @ctz alluded to in the previous PR, we should change the key schedule to use session types. I think this will reduce the memory consumption here. If we need an immediate avoidance of the size increase, we could store the whole key schedule as a `Box<KeySchedule>` instead.

And/or, we could look into creating compressed key types in *ring*, which store just the raw key values instead of the expanded key. That wouldn't be something I can do immediately though.

Reply:
To expand on why session types help: Really, we should be storing the traffic secrets in the session and the rest of the keys should be stored in the handshake. That way, when we finish the handshake we'll free the memory for all the non-traffic keys in the key schedule. This should shrink the overall size (heap + non-heap) of the session.

This also illustrates why a compressed `Prk` type would be useful to add to *ring*: especially after the handshake, we'll rarely use the traffic secret, so we should store them in the minimal form, instead of storing them in a fork that's optimized for frequent use. It will be a while before I do this though. I'm planning to implement bigger and more important size optimizations in *ring* first.

Reply:
Yes, I have a medium-term plan to move more stuff from the `SessionCommon` type into the handshake session types. This also means (eg) post-handshake TLS1.3 sessions don't have any memory used by TLS1.2-specific types.